src/main/java/uk/gov/hmcts/reform/hmc/api/services/PrlUpdateServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.hmc.api.services;
+
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.hmc.api.exceptions.PrlUpdateException;
+import uk.gov.hmcts.reform.hmc.api.model.request.Hearing;
+// import uk.gov.hmcts.reform.hmc.api.model.response.IdamTokenResponse;
+
+@Service
+public class PrlUpdateServiceImpl implements PrlUpdateService {
+
+    // @Autowired private IdamService idamService;
+
+    @Autowired private AuthTokenGenerator authTokenGenerator;
+
+    RestTemplate restTemplate = new RestTemplate();
+
+    @Value("${prl.baseUrl}")
+    private String prlBaseUrl;
+
+    @Value("${prl.path}")
+    private String prlPath;
```

Comment:
no java documents at method level and class levels. pls make sure java best practices